android/app/src/main/java/com/zulipmobile/sharing/ReceiveShareActivity.kt

```diff
@@ -0,0 +1,108 @@
+package com.zulipmobile.sharing
+
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import android.webkit.WebView
+import androidx.annotation.Nullable
+import com.facebook.react.ReactActivity
+import com.facebook.react.ReactApplication
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.Arguments
+import com.zulipmobile.notifications.*
+
+const val TAG = "ZulipReceiveShare"
+
+class ReceiveShareActivity : ReactActivity() {
+
+    /**
+     * Returns the name of the main component registered from JavaScript.
+     * This is used to schedule rendering of the component.
+     */
+    override fun getMainComponentName(): String? {
+        return "SharingRoot"
+    }
+
+    private fun sendEvent(reactContext: ReactContext,
+                          eventName: String,
+                          @Nullable params: WritableMap) {
+        Log.d(TAG, "Sending event with shared data")
+        emit(reactContext, eventName, params)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        WebView.setWebContentsDebuggingEnabled(true)
+        if (intent?.action == Intent.ACTION_SEND) {
+            handleSend(intent)
+        }
+        finish()
+    }
+
+    private fun handleSend(intent: Intent) {
+        val application = application as ReactApplication
+        val host = application.reactNativeHost
+        val reactContext = host.tryGetReactInstanceManager()?.currentReactContext
+        val params: WritableMap
+        try {
+            params = getParamsFromIntent(intent)
+        } catch (e: ShareParamsParseException) {
+            Log.w(TAG, "Ignoring malformed share Intent: ${e.message}")
+            return
+        }
+
+        val appStatus = reactContext?.appStatus
+        when (appStatus) {
+            null, ReactAppStatus.NOT_RUNNING ->
+                // Either there's no JS environment running, or we haven't yet reached
+                // foreground.  Expect the app to check initialSharedData on launch.
+                SharingModule.initialSharedData = params
+            ReactAppStatus.BACKGROUND, ReactAppStatus.FOREGROUND ->
+                // JS is running and has already reached foreground.It won't check
+                // initialSharedData again, but it will see a shareReceived event.
+                sendEvent(reactContext, "shareReceived", params)
+        }
+        when (appStatus) {
+            null, ReactAppStatus.NOT_RUNNING, ReactAppStatus.BACKGROUND ->
+                launchMainActivity(application as Context)
+            ReactAppStatus.FOREGROUND -> Unit
+        }
```

Comment:
For the time being, this code is a close duplicate of the one in `NotifyReact.kt`, until we decide to either make a common function for both or keep the duplicate here.

Reply:
Great. The one essential thing it's missing is a comment (on each copy) mentioning it's duplicated, and pointing to the other copy.

I definitely won't want to keep this code duplicated. But it will be straightforward to factor out.